In [2]:
import numpy as np
from scipy import stats

# Домашнее задание 5   

## Задача 1

> Когда используется критерий Стьюдента, а когда Z –критерий?


Если у нас условия применимости параметрических тестов соблюдены, то в случае известной сигмы генеральной совокупности более предпочтителен Z- критерий, в противном случае -  критерий Стьюдента t

## Задача 2

> Проведите тест гипотезы. Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с $\alpha$=0.05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 мм.

$H0: \mu = 17$  

$H1: \mu > 17$  

Из условий задачи не совсем понятно. Нам дана дисперсия генеральной совокупности или выборки

### а) Если дана дисперсия выборки

$D_{выборки} = 4$  

$\sigma = \sqrt{D_{выборки}}=2$  

$Z_{fact}=\frac{\overline{X}-\mu}{\sigma}$  

$Z_{fact} = \frac{17.5 - 17}{2}=0.25$ 

$\alpha =0.05$

По таблице находим $Z_{crit}$ для 0.95  

$Z_{crit}=1.645$

т.к. $Z_{fact}=0.25 < 1.645= Z_{crit}$  

Альтернативная гипотеза не принимается  
Средний диаметр действительно 17 мм. 

### б) Если дана дисперсия генеральной совокупности

$D_{ген} = 4$  

$\sigma = \sqrt{D_{ген}}=2$  

$Z_{fact}=\frac{\overline{X}-\mu}{\frac{\sigma}{\sqrt{n}}}$  

$Z_{fact} = \frac{17.5 - 17}{2}=2.5$ 

$\alpha =0.05$

По таблице находим $Z_{crit}$ для 0.95  

$Z_{crit}=1.645$

т.к. $Z_{fact}=2.5 >1.645= Z_{crit}$  

Альтернативная гипотеза принимаеся  
Средний диметр больше 17 мм.

## Задача 3
> Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190.
> Известно, что их веса распределены нормально.
>
>- Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?

$H0: M = 200$

$H1: M \not= 200$

$\sigma =\frac{1}{n-1}\sum(X_i-\overline{X})^2$

$t=\frac{\overline{X}-\mu}{\frac{\sigma_н}{\sqrt{n}}}$

In [3]:
X = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])
mean_X = X.mean()
std_X = X.std(ddof=1)

In [4]:
t_fact = (mean_X - 200) / std_X * np.sqrt(10)
t_fact

-1.0651074037450896

По таблице критических точек распределения Стьюдента для двухсторонней области найдем критическую точку по
уровню значимости α = 0,01 и числу степеней свободы k = 9 

$t_{crit}$ = 3.25

$t_{fact}$ = -1.065

$|t_{fact}|<t_{crit}$

$(-3.25<-1.065<3.25)$

Ответ: Утверждение продавца верно

## Задача 4

> Есть ли статистически значимые различия в росте дочерей?  
> Рост матерей 172, 177, 158, 170, 178,175, 164, 160, 169, 165  
> Рост взрослых дочерей: 173, 175, 162, 174, 175, 168, 155, 170, 160

In [5]:
x = np.array([172, 177, 158, 170, 178, 164, 160, 169, 165])
y = np.array([173, 175, 162, 174, 175, 168, 155, 170, 160])
stats.ttest_rel(x,y)

TtestResult(statistic=0.08873565094161137, pvalue=0.9314731778513443, df=8)

$H0:\mu_0=\mu_1$ Различий нет

$H1:\mu_0\not=\mu_1$ Различия есть

$\alpha=0.05$

$P_{value}>\alpha$ Альтернатиная гипотеза не принимается

Статистически значимых различий в росте дочерей нет